### ISSUE TITLE
Watch folders for new/changed/removed photos

### ISSUE THREAD

**Issue by SmilyOrg** [SmilyOrg/photofield#73]
**Is your feature request related to a problem? Please describe.**
Actually using the app in day-to-day life is not a great experience. You have to rescan all the time, which can take forever. If you haven't rescanned in a while, the newly added photos can also take minutes to hours to finish indexing. Additionally, there are a bunch of caches that aren't invalidated when scanning, even if you refresh the page, which seems broken and is very annoying.

**Describe the solution you'd like**

- [ ] Watch for files in realtime, using https://github.com/fsnotify/fsnotify
- [ ] Handle adding new files
- [ ] Handle removing files
- [ ] Handle file renames
- [ ] Handle folder renames
- [ ] Never (?) delete metadata on files so that e.g. moves do not require reindexing. This is also important in case you add e.g. tags to files
- [ ] Add marking scenes as "stale" if the dependencies (i.e. files) change and refresh them, at least upon page refresh

**Describe alternatives you've considered**
Improving the caching part without adding watch.

**Additional context**

**Comment by ckyb** [comment]
To start, would you accept a simple `-scan` command via the cli?
Something like:
`./photofield -scan <folder>`
`./photofield -scan-all`

This would be similar to hitting "rescan" in the UI, but could be run periodically (or on first import of a bunch of folders).

**Comment by SmilyOrg** [comment]
Sure! That's a good idea!

Do you think it makes sense to be able to specify any folder even if it's not part of a collection?

Should you be able to specify a collection in addition/instead? (A collection can consist of multiple folders)

**Comment by ItsNoted** [comment]
> To start, would you accept a simple `-scan` command via the cli? Something like: `./photofield -scan <folder>` `./photofield -scan-all`
> 
> This would be similar to hitting "rescan" in the UI, but could be run periodically (or on first import of a bunch of folders).

Was this ever added? I was looking for an auto refresh or auto scan feature so I don't have to keep doing it manually. Would like to be able to enable it by default using a environment option if possible.

**Comment by SmilyOrg** [comment]
> > To start, would you accept a simple `-scan` command via the cli? Something like: `./photofield -scan <folder>` `./photofield -scan-all`
> > 
> > This would be similar to hitting "rescan" in the UI, but could be run periodically (or on first import of a bunch of folders).
> 
> Was this ever added? I was looking for an auto refresh or auto scan feature so I don't have to keep doing it manually. Would like to be able to enable it by default using a environment option if possible. Using docker.

No, not yet, but it's a good idea still! I just haven't gotten around to adding it :)

I'm thinking of being able to configure it in the config yaml, maybe globally and per collection. Could be a simple syntax of defining an interval, but could also be crontab syntax.

What do you think?
